src/algebra/order/interval.lean

```diff
@@ -0,0 +1,346 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import algebra.order.group.basic
+import data.option.n_ary
+import data.set.pointwise.basic
+import order.interval
+
+/-!
+# Interval arithmetic
+
+This file defines arithmetic operations on intervals and prove their correctness. Note that this is
```

Comment:
As for correctness: are you planning to prove results like `a ∈ s → b ∈ t → a + b ∈ s + t`?

Reply:
I am split between showing these and defining an "interval operation" as something like
```
structure interval_hom (f : α → β) :=
(to_fun : interval α → interval β)
(mem_map' := ∀ a s, a ∈ s → f a ∈  to_fun s)
```
I guess both is an answer.